geode-core/src/main/java/org/apache/geode/internal/cache/control/RestoreRedundancyRegionResult.java

```diff
@@ -0,0 +1,139 @@
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.control;
+
+import java.io.Serializable;
+
+import org.apache.geode.internal.cache.PartitionedRegion;
+
+/**
+ * Used to calculate and store the redundancy status for a {@link PartitionedRegion}.
+ */
+public class RestoreRedundancyRegionResult implements Serializable {
+  private static final long serialVersionUID = 3407539362166634316L;
+
+  /**
+   * The message prefix to be applied when redundancy was successfully restored for the region used
+   * to create this object.
+   */
+  public static final String SATISFIED_MESSAGE = "Redundancy satisfied for region ";
+
+  /**
+   * The message prefix to be applied when redundancy was not successfully restored for the region
+   * used to create this object.
+   */
+  public static final String NOT_SATISFIED_MESSAGE =
+      "Redundancy NOT satisfied for region ";
+
+  /**
+   * The name of the region used to create this object.
+   */
+  private final String regionName;
+
+  /**
+   * The configured redundancy of the region used to create this object.
+   */
+  private final int desiredRedundancy;
+
+  /**
+   * The actual redundancy of the region used to create this object at time of creation.
+   */
+  private final int actualRedundancy;
+
+  /**
+   * The {@link RedundancyStatus} of the region used to create this object at time of creation.
+   */
+  private final RedundancyStatus status;
+
+  /**
+   * The redundancy status of the region used to create this object at time of creation.
+   * {@link #SATISFIED} if every bucket in the region has the configured number of redundant copies
+   * {@link #NOT_SATISFIED} if at least one bucket in the region has less than the configured number
+   * of redundant copies
+   * {@link #NO_REDUNDANT_COPIES} if at least one bucket in the region has zero redundant copies and
+   * the region is not configured for zero redundancy
+   */
+  enum RedundancyStatus {
+    SATISFIED,
+    NOT_SATISFIED,
+    NO_REDUNDANT_COPIES
+  }
+
+  public RestoreRedundancyRegionResult(PartitionedRegion region) {
+    regionName = region.getName();
+    desiredRedundancy = region.getRedundantCopies();
+    actualRedundancy = calculateLowestRedundancy(region);
+    status = determineStatus(desiredRedundancy, actualRedundancy);
+  }
+
+  public String getRegionName() {
+    return regionName;
+  }
+
+  public int getDesiredRedundancy() {
```

Comment:
According to the IDE this method is never invoked.
Unless we foresee somebody using it within the near future, can we just delete it?.

Reply:
Sure